app/controllers/api/base_controller.rb

```diff
@@ -264,6 +244,68 @@ def action_permission
       end
     end
 
+    def parent_permission(child_permission)
+      case child_permission.to_s
+      when 'create', 'destroy'
+        'edit'
+      when 'edit', 'view'
+        'view'
+      else
+        raise ::Foreman::Exception.new(N_("unknown parent permission for %s"), "#{params[:controller]}##{child_permission}")
+      end
+    end
+
+    def parent_resource_details
+      classes_arr = params.map { |key, value| extract_resource_from_param(key, value) }
```

Comment:
When I try you version of the code, Rubocop yells at me that `parent_name, parent_class` are useless assignments.